pkgs/development/compilers/flutter/flutter.nix

```diff
@@ -56,21 +59,36 @@ let
       export STAMP_PATH="$FLUTTER_ROOT/bin/cache/flutter_tools.stamp"
 
       export DART_SDK_PATH="${dart}"
+      export HOME=$PWD
 
-      HOME=../.. # required for pub upgrade --offline, ~/.pub-cache
-                 # path is relative otherwise it's replaced by /build/flutter
-
+      ${dart}/bin/dart pub cache preload .pub-preload-cache/**
       pushd "$FLUTTER_TOOLS_DIR"
-      ${dart}/bin/dart pub get --offline
+        rm -rf test
+        ${dart}/bin/dart pub get --offline
       popd
 
       local revision="$(cd "$FLUTTER_ROOT"; git rev-parse HEAD)"
       ${dart}/bin/dart --snapshot="$SNAPSHOT_PATH" --packages="$FLUTTER_TOOLS_DIR/.dart_tool/package_config.json" "$SCRIPT_PATH"
       echo "$revision" > "$STAMP_PATH"
       echo -n "${version}" > version
 
-      rm -r bin/cache/{artifacts,dart-sdk,downloads}
-      rm bin/cache/*.stamp
+      rm -r bin/cache/dart-sdk
+    '';
+
+    postFixup = ''
```

Comment:
Just use `wrapProgram`?

Reply:
Originally I did try using `wrapProgram`, but didn't succeed as it behaved unexpectedly(such as pointing to incorrect files) and didn't wrap the program.
Additionally, this line `export PUB_CACHE=\''${PUB_CACHE:-"\$HOME/.pub-cache"}\` might be difficult to add with `wrapProgram`.

Maybe someone else will have better luck with `wrapProgram`, but that's my experience..